spec/unit/taric_synchronizer_spec.rb

```diff
@@ -0,0 +1,253 @@
+
+        it 'does not kick off the ClearCacheWorker' do
+          allow(TariffSynchronizer::BaseUpdateImporter).to receive(:perform)
+
+          apply
+
+          expect(Sidekiq::Client).not_to have_received(:enqueue).with(ClearCacheWorker)
+        end
+      end
+
+      context 'when reindex_all_indexes arg is false' do
+        subject(:apply) { described_class.apply(reindex_all_indexes: false) }
+
+        it 'does not kick off the ClearCacheWorker' do
+          allow(TariffSynchronizer::BaseUpdateImporter).to receive(:perform)
+
+          apply
+
+          expect(Sidekiq::Client).not_to have_received(:enqueue).with(ClearCacheWorker)
+        end
+      end
+
+      context 'when reindex_all_indexes arg is true' do
+        subject(:apply) { described_class.apply(reindex_all_indexes: true) }
+
+        it 'kicks off the ClearCacheWorker' do
+          allow(TariffSynchronizer::BaseUpdateImporter).to receive(:perform)
+          allow(TariffSynchronizer::BaseUpdate).to receive(:pending_or_failed).and_return([])
+
+          apply
+
+          expect(Sidekiq::Client).to have_received(:enqueue).with(ClearCacheWorker)
+        end
+      end
+    end
+
+    context 'when unsuccessful' do
+      before do
+        applied_update
+        pending_update
+
+        allow(TariffSynchronizer::BaseUpdateImporter).to receive(:perform).with(pending_update).and_raise(Sequel::Rollback)
+        allow(Sidekiq::Client).to receive(:enqueue)
+      end
+
+      it 'after an error next record is not processed' do
+        expect { described_class.apply }.to raise_error(Sequel::Rollback)
+
+        expect(Sidekiq::Client).not_to have_received(:enqueue).with(ClearCacheWorker)
+      end
+    end
+
+    context 'with failed updates present' do
+      let(:failed_update) { create(:taric_update, :failed, example_date: Time.zone.yesterday) }
+
+      before do
+        failed_update
+      end
+
+      it 'does not apply pending updates' do
+        allow(TariffSynchronizer::TaricUpdate).to receive(:pending_at)
+
+        expect { described_class.apply }.to raise_error(BaseSynchronizer::FailedUpdatesError)
+
+        expect(TariffSynchronizer::TaricUpdate).not_to have_received(:pending_at)
+      end
+
+      it 'logs the error event' do
+        expect_any_instance_of(TariffSynchronizer::Logger).to receive(:failed_updates_present)
+
+        expect { described_class.apply }.to raise_error(BaseSynchronizer::FailedUpdatesError)
+      end
+
+      it 'sends email with the error' do
+        expect { described_class.apply }.to raise_error(BaseSynchronizer::FailedUpdatesError)
+      end
+
+      context 'when reindex_all_indexes arg is true' do
+        subject(:apply) { described_class.apply(reindex_all_indexes: true) }
+
+        it 'does not kick off the ClearCacheWorker' do
+          allow(Sidekiq::Client).to receive(:enqueue)
+          allow(TariffSynchronizer::BaseUpdateImporter).to receive(:perform)
+
+          apply
+        rescue StandardError
```

Comment:
maybe we can split this into two different tests, and avoid to use  rescue inside the test?

- expect to raise StandardError
- 
- expect(Sidekiq::Client).not_to have_received(:enqueue).with(ClearCacheWorker)

Reply:
I think we've had this convo before :thinking: . So I think I'll probably completely rewrite the tests at some point so we can tackle that there

Reply:
ok, yes, we refactored something similar before.